notifications_utils/field.py

```diff
@@ -75,6 +79,8 @@ def __init__(
 
         self.sanitizer = self.get_sanitizer(html)
         self.redact_missing_personalisation = redact_missing_personalisation
+        if "??" in self.content:
+            self.placeholder_pattern = re.compile(r"\({2}([\s\S]+?)\){2}")
```

Comment:
This is a good pattern and I wondered if we could not reuse this one globally instead. I realized why we can't and need to test with the condition instead (for cases with triple parentheses). I wondered what happens if we have a template with both conditional variable and a regular variable and that make that logic fail with the following input (which I pushed to the branch as a test):

```
(((warning))) and ((alert?? alert!))
```

There is a global pattern we can use though by adding a negative lookahead assertion check to your new regex. I will drop it as a suggestion on the pattern at line 57. We could remove this condition here if we adopt it and cover all test cases we have (so far).